tests/trainer/test_trainer.py

```diff
@@ -2146,3 +2146,26 @@ def test_dataloaders_are_not_loaded_if_disabled_through_limit_batches(running_st
         else getattr(trainer, f"{dl_prefix}_dataloaders")
     )
     assert dl is None
+
+
+@pytest.mark.parametrize(
+    ["trainer_kwargs", "expected_device_ids"],
```

Comment:
Can we test num_devices as well please?

Reply:
assert len(trainer.device_ids) == trainer.num_devices should already test it. See L2171

Reply:
Oh  I see! nice!